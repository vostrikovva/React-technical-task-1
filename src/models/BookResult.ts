import Doctor from './Doctor'

export default interface BookResult {
  doctor: Doctor
  createdAt: string
  date: string
}
