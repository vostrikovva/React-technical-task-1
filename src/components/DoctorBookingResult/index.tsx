import { FC } from 'react'
import { Card } from 'react-bootstrap'
import moment from 'moment'

import styles from './styles.module.css'
import BookResult from '../../models/BookResult'
import DateFormats from '../../constants/DateFormats'

const DoctorBookingResult: FC<BookResult> = ({ doctor, createdAt, date }) => {
  const { name, address } = doctor

  return (
    <Card className={styles.root}>
        <Card.Body>
        <Card.Title className="mb-3">You have an appointment with {name} on {moment(date).format(DateFormats.doctorBooking)}</Card.Title>
        {address &&
            <div className="row mb-10">
            <div className="col-auto d-flex align-items-start" >
                <Card.Subtitle>Address</Card.Subtitle>
            </div>
            <div className="col">
                <Card.Text className="mb-0">{`${address.country}, ${address.city}`}</Card.Text>
                <Card.Text className="mb-0">Postal Code: {address.postalCode}</Card.Text>
                <Card.Text className="mb-0">Address line 1: {address.line1}</Card.Text>
                <Card.Text className="mb-0">Address line 2: {address.line2}</Card.Text>
            </div>
            </div>
        }
        <div className="mt-2 d-flex justify-content-end">
            Created at: {moment(createdAt).format(DateFormats.doctorBooking)}
        </div>
        </Card.Body>
    </Card>
  )
}

export default DoctorBookingResult
