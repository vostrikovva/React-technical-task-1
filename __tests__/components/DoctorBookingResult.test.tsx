import { render, cleanup } from '@testing-library/react'
import renderer from 'react-test-renderer'
import '@testing-library/jest-dom'

import DoctorBookingResult from '../../src/components/DoctorBookingResult'
import BookResult from '../../src/models/BookResult'

const bookResult: BookResult = {
  doctor: {
    id: 'some-id-here',
    name: 'John Snow',
    address: {
      line1: "6037 Tonnin de l'Odéon",
      line2: 'Suite 363',
      postalCode: '98950',
      city: 'Port Sébastien',
      country: 'GF'
    }
  },
  createdAt: '2022-10-22 22:33:13',
  date: '2022-10-25 12:30:00'
}

describe('DoctorBookingResult component', () => {
  afterEach(() => {
    cleanup()
  })

  test('Render the component', async () => {
    render(<DoctorBookingResult {...bookResult} />)
  })

  test('Snapshot test', () => {
    const tree = renderer.create(<DoctorBookingResult {...bookResult} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
