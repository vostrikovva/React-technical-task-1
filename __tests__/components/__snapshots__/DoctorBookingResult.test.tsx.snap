// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DoctorBookingResult component Snapshot test 1`] = `
<div
  className="root card"
>
  <div
    className="card-body"
  >
    <div
      className="mb-3 card-title h5"
    >
      You have an appointment with 
      John Snow
       on 
      2022-10-25 12:30:00
    </div>
    <div
      className="row mb-10"
    >
      <div
        className="col-auto d-flex align-items-start"
      >
        <div
          className="card-subtitle h6"
        >
          Address
        </div>
      </div>
      <div
        className="col"
      >
        <p
          className="mb-0 card-text"
        >
          GF, Port Sébastien
        </p>
        <p
          className="mb-0 card-text"
        >
          Postal Code: 
          98950
        </p>
        <p
          className="mb-0 card-text"
        >
          Address line 1: 
          6037 Tonnin de l'Odéon
        </p>
        <p
          className="mb-0 card-text"
        >
          Address line 2: 
          Suite 363
        </p>
      </div>
    </div>
    <div
      className="mt-2 d-flex justify-content-end"
    >
      Created at: 
      2022-10-22 22:33:13
    </div>
  </div>
</div>
`;
